packages/nameguard-sdk/src/index.ts

```diff
@@ -365,34 +369,57 @@ export interface ConsolidatedNameGuardReport extends ConsolidatedReport {
   /* The ENSIP-15 normalization status of `name` */
   normalization: Normalization;
 
-  /** Beautified version of `name` */
+  /* Beautified version of `name` */
   beautiful_name: string;
+
+  /* Whether or not the name was inspected. */
+  inspected: boolean;
 }
 
 /**
  * NameGuard report that contains the full results of all `checks` on all `labels` in a name.
  */
-export interface NameGuardReport extends ConsolidatedNameGuardReport {
+export interface AbstractNameGuardReport extends ConsolidatedNameGuardReport {
   /* The results of all checks performed by NameGuard on `name`. */
   checks: CheckResult[];
 
-  /** Details of the inspection of all labels in `name`. */
-  labels: LabelGuardReport[];
+  /** Details of the inspection of all labels in `name`.
+    *
+    * `undefined` if `name` was uninspected and this is an `UninspectedNameGuardReport`.
+  */
+  labels?: LabelGuardReport[];
 
   /**
    * The name considered to be the canonical form of the analyzed `name`.
    *
-   * `null` if and only if the canonical form of `name` is considered to be undefined.
+   * `undefined` if and only if the canonical form of `name` is considered to be undefined or `name` was uninspected and this is an ``UninspectedNameGuardReport`.
    *
    * If a label is represented as `[labelhash]` in `name`,
    * the `canonical_name` will also contain the label represented as `[labelhash]`.
    *
    * `canonical_name` is guaranteed to be normalized with the exception of the case
    * where `normalization` is `unknown` and one or more labels are represented as `[labelhash]`.
    */
-  canonical_name: string | null;
```

Comment:
@djstrong Two ideas here:
1. Can we please document here what cases cause `canonical_name` to be undefined?
2. When we document these cases, can we please include reference to the new `inspected` field?